### ISSUE TITLE
More map checks

### ISSUE THREAD

**Issue by eevee** [sirjuddington/SLADE#383]
Some obvious errors that come to mind:

* [ ] A thing with no skill level flags, or none of the SP/DM/CO flags
* [ ] A dormant thing/monster with no tid
* [ ] A door track that's not lower unpegged
* [ ] A locked special with no corresponding key (taking into account skill/MP flags)
* [ ] Missing textures in places that only become visible during play
* [ ] Reference to a script (on the current map, at least) that doesn't exist
* [ ] A line with a non-init special but no triggers; or with an init special and triggers; or with triggers but no special; or with a repeatable/etc flag but no triggers (maybe only if the line has no lineid)
* [ ] Specials with tags that don't appear on the map
* [ ] Translucency on a 1-sided line

And since you might actually want to do some of those things in a few perverse cases, maybe also:

* [ ] Allow permanently ignoring a particular map check on a particular object via a custom UDMF property

**Comment by jmickle66666666** [comment]
Yeah I think all map errors like this should be toggleable, both as a setting in Slade and in the map data. If it's only as a UDMF setting then users won't be able to toggle them in other map formats. The door track pegging would be a useful one but is also used very often on purpose.

**Comment by eevee** [comment]
The map checks panel already has a checkbox for each individual type of check, so if you _really really_ want pegged doors, you can just turn that check off.

**Comment by Xeriphas1994** [comment]
Hope it's OK to tag along here, as a longtime user of an app with excellent checks but no bus factor. :)
- [ ] Thing not in any sector
- [ ] Overlapping pickups on same skill level
- [ ] Non-matching Icon of Sin components
- [ ] Medusa effect
- [ ] Texture offset exceeds dimension of graphic
- [ ] Vertex not at the ends of lines
- [ ] Overlapping vertexes
- [ ] Duplicate sidedef IDs (if known not to be packed)
- [ ] Linedef 0 likely to overflow blocklist operation
- [ ] Line shorter than some minimum
- [ ] Tag present but not required
- [ ] Dubious stair builder (inconsistent tagging, missing textures, different complevel than currently selected)

A port developer told me that finding all missing/unnecessary textures is impossible, because the number of sight lines grows exponentially with the number of moving sectors. (I assume scripting makes it worse, because you have to traverse control structures to even enumerate the possible positions?)

Some checks were originally responding to known bugs in 1990s editors -- which have suddenly become important again with the "Bethesda Edition" collection! :D
